resources/scripts/components/server/features/JavaVersionModalFeature.tsx

```diff
@@ -0,0 +1,110 @@
+    const { connected, instance } = ServerContext.useStoreState(state => state.socket);
+    let selectedVersion = 'ghcr.io/pterodactyl/yolks:java_16';
+
+    const dockerImageList = [
+        { name: 'Java 8', image: 'ghcr.io/pterodactyl/yolks:java_8' },
+        { name: 'Java 11', image: 'ghcr.io/pterodactyl/yolks:java_11' },
+        { name: 'Java 16', image: 'ghcr.io/pterodactyl/yolks:java_16' },
+    ];
+
+    useEffect(() => {
+        if (!connected || !instance || status === 'running') return;
+
+        const errors = [ 'minecraft 1.17 requires running the server with java 16 or above',
+            'java.lang.unsupportedclassversionerror',
+            'unsupported major.minor version',
+            'has been compiled by a more recent version of the java runtime' ];
+
+        const listener = (line: string) => {
+            if (errors.some(p => line.toLowerCase().includes(p))) {
+                setVisible(true);
+            }
+        };
+
+        instance.addListener(SocketEvent.CONSOLE_OUTPUT, listener);
+
+        return () => {
+            instance.removeListener(SocketEvent.CONSOLE_OUTPUT, listener);
+        };
+    }, [ connected, instance, status ]);
+
+    const selectVersion = (event: { target: { value: string; }; }) => {
+        selectedVersion = event.target.value;
+    };
+
+    const updateJava = () => {
+        setLoading(true);
+        clearFlashes('feature:javaversion');
+
+        setSelectedDockerImage(uuid, selectedVersion)
+            .then(() => {
+                if (status === 'offline' && instance) {
+                    instance.send(SocketRequest.SET_STATE, 'restart');
+                }
+
+                setLoading(false);
+                setVisible(false);
+            })
+            .catch(error => {
+                console.error(error);
+                clearAndAddHttpError({ key: 'feature:javaversion', error });
+            })
+            .then(() => setLoading(false));
+    };
+
+    useEffect(() => () => {
+        clearFlashes('feature:javaversion');
+    }, []);
+
+    return (
+        !visible ?
+            null
+            :
+            <Modal visible onDismissed={() => setVisible(false)} closeOnBackground={false} showSpinnerOverlay={loading}>
+                <FlashMessageRender key={'feature:javaversion'} css={tw`mb-4`}/>
+                <h2 css={tw`text-2xl mb-4 text-neutral-100`}>Invalid Java Version, Update Docker Image?</h2>
+                <p css={tw`mt-4`}>This server is unable to start due to the required java version not being met.</p>
+                <p css={tw`mt-4`}>By pressing {'"Update Docker Image"'} below you are acknowledging that the docker image this server uses will be changed to a image below that has the Java version you are requesting.</p>
+                <div css={tw`sm:flex items-center mt-4`}>
+                    <p>Please select a Java version from the list below.</p>
+                    <Select
+                        onChange={selectVersion}
```

Comment:
Change this to use `setSelectedVersion()` directly.

`onChange={e => setSelectedVersion(e.target.value)}`

Reply:
Fixed in next commit